File: DevExpress VCL/ExpressBars 7/Demos/Delphi/RibbonNotepadDemo/RibbonNotepadDemoOptions.pas
unit RibbonNotepadDemoOptions;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, cxGraphics, cxControls, cxLookAndFeels, ExtCtrls;

type
  TRibbonDemoStyle = (rdsOffice2007, rdsOffice2010, rdsScenic);

  TScreenTipOptions = record
    ShowScreenTips: Boolean;
    ShowDescripitons: Boolean;
  end;

  { TRibbonDemoOptionsForm }

  TRibbonDemoOptionsForm = class(TForm)
    Button1: TButton;
    Button2: TButton;
    gbPanel: TBevel;
    cbColorScheme: TComboBox;
    cbRibbonStyle: TComboBox;
    cbScreenTipStyle: TComboBox;
    lblColorScheme: TLabel;
    lblRibbonStyle: TLabel;
    lblScreenTipStyle: TLabel;
  public
    procedure LoadOptions(const AColorSchemeName: string;
      const AScreenTipOptions: TScreenTipOptions; AStyle: TRibbonDemoStyle);
    procedure SaveOptions(out AColorSchemeName: string;
      out AScreenTipOptions: TScreenTipOptions; out AStyle: TRibbonDemoStyle);
  end;

function ExecuteRibbonDemoOptions(var AColorSchemeName: string;
  var AScreenTipOptions: TScreenTipOptions; var AStyle: TRibbonDemoStyle): Boolean;
implementation

{$R *.dfm}

function ExecuteRibbonDemoOptions(var AColorSchemeName: string;
  var AScreenTipOptions: TScreenTipOptions; var AStyle: TRibbonDemoStyle): Boolean;
begin
  with TRibbonDemoOptionsForm.Create(nil) do
  try
    LoadOptions(AColorSchemeName, AScreenTipOptions, AStyle);
    Result := ShowModal = mrOk;
    if Result then
      SaveOptions(AColorSchemeName, AScreenTipOptions, AStyle);
  finally
    Free;
  end;
end;

{ TRibbonDemoOptionsForm }

procedure TRibbonDemoOptionsForm.LoadOptions(const AColorSchemeName: string;
  const AScreenTipOptions: TScreenTipOptions; AStyle: TRibbonDemoStyle);
begin
  cbRibbonStyle.ItemIndex := Ord(AStyle);
  cbColorScheme.ItemIndex := cbColorScheme.Items.IndexOf(AColorSchemeName);
  if AScreenTipOptions.ShowScreenTips then
    cbScreenTipStyle.ItemIndex := Ord(not AScreenTipOptions.ShowDescripitons)
  else
    cbScreenTipStyle.ItemIndex := 2;
end;

procedure TRibbonDemoOptionsForm.SaveOptions(out AColorSchemeName: string;
  out AScreenTipOptions: TScreenTipOptions; out AStyle: TRibbonDemoStyle);
begin
  AColorSchemeName := cbColorScheme.Text;
  AStyle := TRibbonDemoStyle(cbRibbonStyle.ItemIndex);
  AScreenTipOptions.ShowScreenTips := cbScreenTipStyle.ItemIndex <> 2;
  AScreenTipOptions.ShowDescripitons := cbScreenTipStyle.ItemIndex = 0;
end;

end.

File: DevExpress VCL/ExpressBars 7/Demos/CBuilder/RibbonNotepadDemo/RibbonNotepadDemoOptions.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "RibbonNotepadDemoOptions.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TRibbonDemoOptionsForm *RibbonDemoOptionsForm;

BOOL ExecuteRibbonDemoOptions(String *AColorSchemeName,
	TScreenTipOptions *AScreenTipOptions, TRibbonDemoStyle *AStyle)
{
	TRibbonDemoOptionsForm *Form = new TRibbonDemoOptionsForm(NULL);
	Form->LoadOptions(*AColorSchemeName, *AScreenTipOptions, *AStyle);
	BOOL Result = Form->ShowModal() == mrOk;
	if (Result)
	  Form->SaveOptions(AColorSchemeName, AScreenTipOptions, AStyle);
	delete Form;
	return Result;
}

//---------------------------------------------------------------------------
__fastcall TRibbonDemoOptionsForm::TRibbonDemoOptionsForm(TComponent* Owner)
		: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void TRibbonDemoOptionsForm::LoadOptions(String AColorSchemeName,
	TScreenTipOptions AScreenTipOptions, TRibbonDemoStyle AStyle)
{
	cbRibbonStyle->ItemIndex = (Integer)AStyle;
	cbColorScheme->ItemIndex = cbColorScheme->Items->IndexOf(AColorSchemeName);
	if (AScreenTipOptions.ShowScreenTips)
		cbScreenTipStyle->ItemIndex = (Integer)(!AScreenTipOptions.ShowDescripitons);
	else
    	cbScreenTipStyle->ItemIndex = 2;
}
//---------------------------------------------------------------------------
void TRibbonDemoOptionsForm::SaveOptions(String *AColorSchemeName,
	TScreenTipOptions *AScreenTipOptions, TRibbonDemoStyle *AStyle)
{
	* AColorSchemeName = cbColorScheme->Text;
	* AStyle = (TRibbonDemoStyle)cbRibbonStyle->ItemIndex;
	AScreenTipOptions->ShowScreenTips = cbScreenTipStyle->ItemIndex != 2;
	AScreenTipOptions->ShowDescripitons = cbScreenTipStyle->ItemIndex == 0;
}
//---------------------------------------------------------------------------
